app/models/events/pull_request.rb

```diff
@@ -23,15 +23,13 @@
 #
 module Events
   class PullRequest < ApplicationRecord
-    ACTIONS = %w[open review_requested closed \
-                 review_request_removed].freeze
-    private_constant :ACTIONS
-
     enum state: { open: 'open', closed: 'closed' }
 
```

Comment:
This is a style question that applies to all model definitions as well.

Would it be possible to add the declaration of each attribute to the class definition?

Even though it is not necessary the rationale is the class to self document its attributes and allow a reader of the code to know at a glance which attributes does an ActiveRecord subclass has without looking into the migration, the database schema or the tests.

It would also declare each attribute configuration like being nullable or having a default value.

Reply:
What do you mean? I didn't get the point

Reply:
I meant to declare not only the model relations but also the model attributes:

```
class Event < ApplicationRecord
  TYPES = %w[pull_request review review_comment].freeze

  attribute :data, default: {}
  attribute :handleable_type
  attribute :name
  attribute :type

  belongs_to :handleable, polymorphic: true, optional: true
  belongs_to :project
  validates :name, :data, presence: true
end
```

But as I said before probably it is related to a code style.

Reply:
This does not apply in Rails